### PR TITLE
fix(github): strip planner-injected search qualifiers from topic queries

### LINKED ISSUE #949: GitHub source: planner-injected qualifiers create duplicate created: — fetches results, reports zero
## Summary

The GitHub source silently returns **0 items on every run** when the planner writes search qualifiers into the topic string (e.g. `open source AI stars:>1000 created:>2025-03-20`). Distinct from #916 (missing `is:` qualifiers) — this is a date-window collision.

## Mechanism

1. The planner emits a topic containing qualifiers: `open source AI stars:>1000 created:>2025-03-20`.
2. `extract_core_subject()` (`query.py`) does not strip qualifiers.
3. `search_github()` (`lib/github.py`) builds `q = "{core} created:>{from_date}"` — producing **two** `created:` qualifiers.
4. GitHub's API honors the **first** (`created:>2025-03-20`) and silently ignores the appended 30-day window. Verified against the API directly: items returned dated 2025-05…2026-05.
5. The local date filter then drops all fetched items. Emitted as `source_status: {"github": "no-results"}` — not an error, so nothing surfaces to the user.

## Repro (isolated on v3.18.3, still present in v3.18.4)

```
CLEAN topic   -> raw: 10  after filter: 10
PLANNER topic -> raw: 10  after filter: 0
```

`planner.py`, `query.py`, and this path in `github.py` are unchanged between 3.18.3 and 3.18.4, so the current release still has it.

## Fix (tested)

Strip GitHub search qualifiers from the topic before the query is built, and error explicitly on qualifier-only topics instead of searching the whole site. Patch below — with it applied: same failing input goes 0 -> 10 items kept, all in-window; full test suite exit 0; `"AI in healthcare"` survives intact (plain word `in` is not treated as the `in:` qualifier).

```diff
--- a/skills/last30days/scripts/lib/github.py	2026-07-26 18:53:16
+++ b/skills/last30days/scripts/lib/github.py	2026-08-01 20:22:35
@@ -166,6 +166,38 @@
     return round(relevance, 2)
 
 
+# GitHub search qualifiers the planner sometimes writes straight into the topic
+# string (e.g. "open source AI stars:>1000 created:>2025-03-20"). They must not
+# reach the query builder in `search_github`: it appends its own
+# `created:>{from_date}`, and when two `created:` qualifiers collide GitHub
+# honours the FIRST and silently ignores ours. The API then returns
+# out-of-window items that `parse_github_response`'s date filter drops
+# wholesale — a source that fetches results and reports zero.
+QUALIFIER_KEYS = frozenset({
+    "archived", "assignee", "author", "base", "closed", "comments", "commenter",
+    "created", "fork", "forks", "head", "in", "interactions", "involves", "is",
+    "label", "language", "license", "linked", "mentions", "merged", "milestone",
+    "no", "org", "project", "pushed", "reactions", "repo", "review",
+    "review-requested", "reviewed-by", "size", "sort", "stars", "state", "team",
+    "topic", "topics", "type", "updated", "user",
+})
+
+_QUALIFIER_RE = re.compile(
+    r"(?:(?<=\s)|^)(?:" + "|".join(sorted(QUALIFIER_KEYS)) + r"):(?:[<>]=?)?\S*",
+    re.IGNORECASE,
+)
+
+
+def strip_search_qualifiers(text: str) -> str:
+    """Strip GitHub search qualifiers from a topic, leaving plain-language text.
+
+    Whitespace is collapsed. Returns an empty string when the topic was nothing
+    but qualifiers; callers must handle that rather than searching on an empty
+    term, which would match the entire site.
+    """
+    return " ".join(_QUALIFIER_RE.sub(" ", text).split())
+
+
 def search_github(
     topic: str,
     from_date: str,
@@ -193,6 +225,21 @@
     """
     count = DEPTH_LIMITS.get(depth, DEPTH_LIMITS["default"])
     core = extract_core_subject(topic)
+    plain_core = strip_search_qualifiers(core)
+    if plain_core != core:
+        _log(f"Stripped search qualifiers: '{core}' -> '{plain_core}'")
+    if not plain_core:
+        # A qualifier-only topic leaves nothing to search on. Report it instead
+        # of querying an empty term, which would match the whole site and then
+        # be discarded by the date filter as a bogus "no results".
+        _log("Topic contained only search qualifiers; nothing to search")
+        return {
+            "items": [],
+            "context": {"core": core, "from_date": from_date,
+                        "to_date": to_date, "count": count},
+            "error": f"GitHub topic contained only search qualifiers: {topic!r}",
+        }
+    core = plain_core
     resolved_token = _resolve_token(token)
     authed = bool(resolved_token)
     if not authed:

```

### PR DESCRIPTION
**Pull request by SomSamantray**
## Summary

The GitHub source silently returned 0 items for any topic where the planner wrote search qualifiers (e.g. `open source AI stars:>1000 created:>2025-03-20`). The query builder appended its own `created:>{from_date}`, GitHub honored the first `created:` qualifier and ignored the 30-day window, and the local date filter then dropped every out-of-window item — a source that fetched results and reported zero. Qualifiers are now stripped from the topic before the query is built (including comma/semicolon-glued and quoted-value forms such as `label:"bug fix"`), and a qualifier-only topic reports an explicit error instead of searching the whole site.

## Testing

- [x] `uv run pytest` — full suite green, including 13 new GitHub tests
- [x] New tests: unit coverage for `strip_search_qualifiers` (plain words survive; qualifier-key words without a colon survive; case-insensitive; comma/semicolon-glued and quoted values consumed) and end-to-end `search_github` assertions that the built query carries exactly one `created:` qualifier, that qualifier-only/empty topics error without a network call, and that no `stars:`/`created:>...` remnants reach the query
- [x] The new test classes were appended after the `unittest.main()` guard in the first commit; they were moved to the end of the file so direct `python tests/test_github.py` runs execute them (verified 35 -> 43 tests)

## Changelog

- [x] Added `changelog.d/949.fixed.md`

### AI review

The change was reviewed by four personas (correctness, testing, project-standards, adversarial). Two edge-case defects were found and fixed before shipping: comma-glued qualifiers (`ai,created:>...`) and quoted qualifier values (`label:"bug fix"`) bypassed the original strip boundary, both independently reproduced. Report-only residual findings (wrapped qualifier shapes, qualifier-only ERROR classification, per-subquery log noise) were filed as issues #951-#954.

### Security

N/A — no input handling, command execution, path handling, auth, secrets, or dependency changes.

## Notes

Qualifier stripping is scoped to the GitHub adapter; the shared `extract_core_subject()` in `query.py` is untouched. Session-settled decisions carried from planning: strip GitHub search qualifiers before query build (user-approved, over leaving the silent no-results behavior as-is); scope the fix to the GitHub source adapter (user-directed, over changing the shared query path). Residual review concerns are tracked in #951-#954.

### Relationship to this change

- [x] None
- [ ] Yes — disclosure: 

## Related issues

Fixes #949

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

**Comment by SomSamantray** [comment]
Addressed in 4f417cc: the qualifier-value class now stops at separators instead of greedily consuming the suffix, so topic terms glued after a qualifier value (e.g. `created:>2025-03-20,robotics`) are preserved. Unit and adapter-level tests added for the glued-term-after-value case.
